_posts/2023-01-31-tuf.md

```diff
@@ -0,0 +1,779 @@
+  | Keys      | ------> | Metadata            | ------------> | Metadata      |
+  +-----------+         | 1.root.json         |               | root.json     |  
+  | Metadata  |         | 1.targets.json      |               | timestamp.json|
+  | root.json |         | 1.snapshot.json     |               +---------------+
+  +-----------+         | timestamp.json      |
+  | thing_v1  |         +---------------------+
+  +-----------+         | thing_v1            |
+                        +---------------------+
+```
+And there will number of verifications performed on timestamp.json.
+
+Next, the client will [download](https://theupdateframework.github.io/specification/latest/index.html#update-snapshot) the snapshot metadata file:
+```
+  +-----------+         +---------------------+               +---------------+
+  | Producer  |         | TUF Repository      | <-----------  | TUF Consumer  |
+  |-----------+ update  |---------------------+               |---------------|
+  | Keys      | ------> | Metadata            | ------------> | Metadata      |
+  +-----------+         | 1.root.json         |               | root.json     |  
+  | Metadata  |         | 1.targets.json      |               | timestamp.json|
+  | root.json |         | 1.snapshot.json     |               | snapshot.json |
+  +-----------+         | timestamp.json      |               +---------------+
+  | thing_v1  |         +---------------------+
+  +-----------+         | thing_v1            |
+                        +---------------------+
+```
+And there will number of verifications performed on snapshot.json.
+
+Next, the client will [download](https://theupdateframework.github.io/specification/latest/index.html#update-targets) the targets.json metadata file:
+```
+  +-----------+         +---------------------+               +---------------+
+  | Producer  |         | TUF Repository      | <-----------  | TUF Consumer  |
+  |-----------+ update  |---------------------+               |---------------|
+  | Keys      | ------> | Metadata            | ------------> | Metadata      |
+  +-----------+         | 1.root.json         |               | root.json     |  
+  | Metadata  |         | 1.targets.json      |               | timestamp.json|
+  | root.json |         | 1.snapshot.json     |               | snapshot.json |
+  +-----------+         | timestamp.json      |               | targets.json  |
+  | thing_v1  |         +---------------------+               +---------------+
+  +-----------+         | thing_v1            |
+                        +---------------------+
+```
+And there will number of verifications performed on targets.json.
+
+Finally, the client will [fetch](https://theupdateframework.github.io/specification/latest/index.html#fetch-target) the actual target files, these are the actual artifacts.
+
+The names of the metadata files are named after four top level roles in TUF, 
+the `Root`, `Target`, `Timestamp`, and `Snapshot` roles.
```

Comment:
Maybe put a short bullet point list with a high level description of each role here before diving into the specifics of the roles.